assignment-1/Models/Library.cs

```diff
@@ -0,0 +1,241 @@
+public class Library
+{
+    private List<Book> books;
+    private List<User> users;
+    private List<Admin> admins;
+
+    public Library()
+    {
+        books = new List<Book>();
+        users = new List<User>();
+        admins = new List<Admin>();
+    }
+
+    public void CreateAndAddAdmin(string username, string password)
+    {
+        Admin admin = new Admin(username, password);
+        admins.Add(admin);
+    }
+
+    public User FindUser(string name)
+    {
+        return users.Find(user => user.Name == name);
+    }
+
+    public Book FindBook(string title)
+    {
+        return books.Find(book => book.Title == title);
+    }
+
+    public void AddBook(string adminUsername, string adminPassword, Book book)
+    {
+        Admin admin = FindAdmin(adminUsername, adminPassword);
+        if (admin == null)
+        {
+            Console.WriteLine("\nAdmin authentication failed.");
+            return;
+        }
+
+        books.Add(book);
+        Console.WriteLine($"\nAdded book: {book.Title}");
+    }
+
+    public void RegisterUser(User user)
+    {
+        users.Add(user);
+        Console.WriteLine($"\nRegistered user: {user.Name}");
+    }
+
+    public void IssueBook(string adminUsername, string adminPassword, User user, Book book)
+    {
+        Admin admin = FindAdmin(adminUsername, adminPassword);
+        if (admin == null)
+        {
+            Console.WriteLine("\nAdmin authentication failed.");
+            return;
+        }
+
+        if (user.UserType == UserType.Teacher && user.IssuedBooks.Count >= Constants.MaxBooksPerTeacher)
+        {
+            Console.WriteLine("\nTeacher has already reached maximum issuance limit.");
+            return;
+        }
+        else if (user.UserType == UserType.Student && user.IssuedBooks.Count >= Constants.MaxBooksPerUser)
+        {
+            Console.WriteLine("\nStudent has already reached maximum issuance limit.");
+            return;
+        }
+
+        if (book.InventoryCount == 0)
+        {
+            Console.WriteLine("\nBook is out of stock.");
+            return;
+        }
+
+        user.IssuedBooks.Add(book);
+        book.IssuedTo.Add(user);
+        book.IssueDate = DateTime.Now;
+        book.InventoryCount--;
+        user.ReturnDate = DateTime.Now.AddDays(7);

```

Comment:
There should be a constant storing this `7` and you should use it here. To make this even more readable you can create constants for all the Menu options and response messages